File: backend/timebank/src/main/java/kookmin/software/capstone2023/timebank/application/service/user/UserFinder.java
package kookmin.software.capstone2023.timebank.application.service.user;

import kookmin.software.capstone2023.timebank.domain.model.User;
import kookmin.software.capstone2023.timebank.domain.repository.UserJpaRepository;
import org.springframework.stereotype.Service;

@Service
public class UserFinder {

    private final UserJpaRepository userJpaRepository;

    public UserFinder(UserJpaRepository userJpaRepository) {
        this.userJpaRepository = userJpaRepository;
    }

    public User findById(Long userId) {
        return userJpaRepository.findById(userId).orElse(null);
    }
}

File: src/main/java/kookmin/software/capstone2023/timebank/application/service/user/UserFinder.kt
package kookmin.software.capstone2023.timebank.application.service.user

import kookmin.software.capstone2023.timebank.domain.model.User
import kookmin.software.capstone2023.timebank.domain.repository.UserJpaRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class UserFinder(
    private val userJpaRepository: UserJpaRepository,
) {
    fun findById(userId: Long): User? {
        return userJpaRepository.findByIdOrNull(userId)
    }
}
